subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/accessors/PluginAccessorsClassPath.kt

```diff
@@ -83,18 +97,114 @@ fun pluginSpecBuildersClassPath(project: Project): AccessorsClassPath = project.
 
     rootProject.getOrCreateProperty("gradleKotlinDsl.pluginAccessorsClassPath") {
         val buildSrcClassLoaderScope = baseClassLoaderScopeOf(rootProject)
-        val cacheKeySpec = accessorsCacheKeySpecPrefix + buildSrcClassLoaderScope.exportClassLoader
-        cachedAccessorsClassPathFor(rootProject, cacheKeySpec) { srcDir, binDir ->
-            kotlinScriptClassPathProviderOf(rootProject).run {
-                withAsynchronousIO(rootProject) {
-                    buildPluginAccessorsFor(
-                        pluginDescriptorsClassPath = exportClassPathFromHierarchyOf(buildSrcClassLoaderScope),
-                        srcDir = srcDir,
-                        binDir = binDir
-                    )
-                }
+        val workExecutor = rootProject.serviceOf<WorkExecutor<ExecutionRequestContext, CachingResult>>()
+        val sourcesOutputDir : File = TODO()
+        val classesOutputDir : File = TODO()
+        val work = GeneratePluginAccessors(
+            rootProject,
+            buildSrcClassLoaderScope,
+            sourcesOutputDir,
+            classesOutputDir,
+            rootProject.serviceOf(),
+            rootProject.serviceOf(),
+            rootProject.serviceOf(),
+            rootProject.serviceOf()
```

Comment:
We could get the services in the constructor instead.

Reply:
@paul Is there a reason why we don't use injection via a `ServiceRegistry` for Kotlin DSL, like we do with other parts of the code? Having a scope dedicated to Kotlin DSL services would make it easier to inject a workspace provider (given that we'll have other workspace providers in other scopes present).

Reply:
The Kotlin DSL uses service injection in general. Here, in `PluginAccessorsClassPath` we are dealing with plain functions, not injectable types, hence the extra gymnastic needed. We could refactor this as a service in order to get injection but I would not do that as part of this work, this can be a cleanup follow up.